pkg/nats/transport/nats.go

```diff
@@ -86,37 +89,50 @@ type NATSTransport struct {
 	nodeInfoDecorator models.NodeInfoDecorator
 }
 
+//nolint:funlen
 func NewNATSTransport(ctx context.Context,
-	config NATSTransportConfig,
-	nodeInfoStore routing.NodeInfoStore) (*NATSTransport, error) {
+	config NATSTransportConfig) (*NATSTransport, error) {
 	log.Debug().Msgf("Creating NATS transport with config: %+v", config)
 	if err := config.Validate(); err != nil {
 		return nil, fmt.Errorf("error validating nats transport config. %w", err)
 	}
 
 	var sm *nats_helper.ServerManager
 	if config.IsRequesterNode {
+		var err error
+
 		// create nats server with servers acting as its cluster peers
-		routes, err := nats_helper.RoutesFromSlice(config.ClusterPeers)
-		if err != nil {
-			return nil, err
+		serverOpts := &server.Options{
+			ServerName:             config.NodeID,
+			Port:                   config.Port,
+			ClientAdvertise:        config.AdvertisedAddress,
+			Authorization:          config.AuthSecret,
+			Debug:                  true, // will only be used if log level is debug
+			JetStream:              true,
+			DisableJetStreamBanner: true,
+			StoreDir:               config.StoreDir,
 		}
-		serverOps := &server.Options{
-			ServerName:      config.NodeID,
-			Port:            config.Port,
-			ClientAdvertise: config.AdvertisedAddress,
-			Routes:          routes,
-			Authorization:   config.AuthSecret,
-			Debug:           true, // will only be used if log level is debug
-			Cluster: server.ClusterOpts{
+
+		// Only set cluster options if cluster peers are provided. If we don't Jetstream
+		// will fail to start as it is unable to locate the zero routes we provide when
+		// there are no peers.
+		if len(config.ClusterPeers) > 0 {
```

Comment:
When ClusterPeers has a value, Jetstream will use those values to create the cluster by attempting to communicate with them.  The problem here is that it's an empty list, and so we specify Clusters as an empty []string.  This causes Jetstream to get into a mess because it thinks it should have a cluster but doesn't know where the nodes are.  I think a potential oversight from nats?

Reply:
I read a little about [JetStream Clustering](https://docs.nats.io/running-a-nats-service/configuration/clustering/jetstream_clustering), and it seems that NATS uses the _defined_ cluster size to configure its RAFT and decide on quorum threshold, which sounds reasonable as we can't expect the requester to build quorum if it doesn't know the size of the cluster beforehand.

What this means for us is every requester must know about the other requesters before they start. This might require changes in how we start our environments. A simple change might be to make to other nodes compute nodes instead of hybrid nodes.